### PR TITLE
Allow use of aws iam role for service accounts

### PR DESCRIPTION
**Pull request by tylerphelan**
- Allows auth with ecr registry using node identity

Requires kpack pr https://github.com/pivotal/kpack/pull/959

### CODE REVIEW

**Comment by matthewmcnew** [comment]
Should the kp_default_repository_aws_iam_role_arn value be added to the kpack controller service account?

File: config/kp-config.yaml [hunk 1]
```diff
@@ -19,10 +19,24 @@
...
-#@  assert.fail("must only use one of kp_default_repository_secret or kp_default_repository creds")
+#@ def exactly_one(values):
+#@     found = False;
+#@     for v in values:
+#@     	if v:
+#@     		if found:
+#@     			return False
+#@     		end
+#@     		found = True
+#@     	end
```
Comment (ncarlson) [inline config/kp-config.yaml:30]: Rather minor, but there's mixing of tabs and spaces

Comment (ncarlson) [inline config/kp-config.yaml:30]: Rather minor, but there's mixing of tabs and spaces

**Comment by tylerphelan** [comment]
> Should the kp_default_repository_aws_iam_role_arn value be added to the kpack controller service account?

This was a mistake, the overlay should be here

**Review by ncarlson — APPROVED** [review]

**Review by ncarlson — APPROVED** [review]

File: config/kp-config.yaml [hunk 2]
```diff
@@ -107,5 +102,34 @@ spec:
         - name: credentials
           secret:
             secretName:  #@ secret_name
+#@ else:
+#@overlay/match by=overlay.subset({"kind": "ServiceAccount", "metadata":{"name":"controller", "namespace":"kpack"}})
```
Comment (matthewmcnew) [inline config/kp-config.yaml:106]: 🎯

**Review by matthewmcnew — APPROVED** [review]

File: config/kp-config.yaml [hunk 3]
```diff
@@ -19,10 +19,24 @@
...
+#@       if found:
+#@         return False
+#@       end
+#@       found = True
+#@     end
+#@   end
+#@   return found
+#@ end
+#@ exclusive_values = [data.values.kp_default_repository_secret.name, data.values.kp_default_repository_username, data.values.kp_default_repository_aws_iam_role_arn]
+#@ if not exactly_one(exclusive_values):
```
Comment (matthewmcnew) [inline config/kp-config.yaml:35]: Would we ever allow both?

Comment (tylerphelan) [reply config/kp-config.yaml:35]: maybe if it gets requested!

**Pull request merged**